contracts/extensions/StakedExpenditure.sol

```diff
@@ -129,17 +132,50 @@ contract StakedExpenditure is ColonyExtensionMeta {
     colony.obligateStake(msgSender(), _domainId, stakeAmount);
     uint256 expenditureId = colony.makeExpenditure(_permissionDomainId, _childSkillIndex, _domainId);
 
-    stakes[expenditureId] = Stake({ creator: msgSender(), amount: stakeAmount });
+    stakes[expenditureId] = Stake({ creator: msgSender(), stagedPayment: _stagedPayment, amount: stakeAmount });
     colony.transferExpenditure(expenditureId, msgSender());
 
-    emit ExpenditureMadeViaStake(msgSender(), expenditureId, stakeAmount);
+    emit ExpenditureMadeViaStake(msgSender(), expenditureId, stakeAmount, _stagedPayment);
+  }
+
+  /// @notice Release a staged payment slot
+  /// @param _permissionDomainId The domainId in which the extension has the arbitration permission
+  /// @param _childSkillIndex The index that the `_expenditureId` is relative to `_permissionDomainId`,
+  /// @param _expenditureId The id of the expenditure
+  /// @param _slot The slot being released
+  /// @param _tokens An array of payment tokens associated with the slot
+  function releaseStagedPayment(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _expenditureId,
+    uint256 _slot,
+    address[] memory _tokens
+  )
+    public
+  {
+    require(stakes[_expenditureId].stagedPayment, "staked-expenditure-not-staged-payment");
+    require(stakes[_expenditureId].creator == msgSender(), "staked-expenditure-not-creator");
+    require(
+      colony.getExpenditure(_expenditureId).status == ColonyDataTypes.ExpenditureStatus.Finalized,
+      "staked-expenditure-not-finalized"
+    );
+
+    bool[] memory mask = new bool[](2); mask[0] = false; mask[1] = true;
+    bytes32[] memory keys = new bytes32[](2); keys[0] = bytes32(0); keys[1] = bytes32(uint256(1));
+    colony.setExpenditureState(_permissionDomainId, _childSkillIndex, _expenditureId, 26, mask, keys, bytes32(0));
+
+    for (uint256 i; i < _tokens.length; i++) {
+      colony.claimExpenditurePayout(_expenditureId, _slot, _tokens[i]);
```

Comment:
Can we get a test demonstrating that it is possible to finalize, and not claim, in this scheme? (This will become important once, for example, you finish the changes to voting and there is an ongoing vote when this function is called)

On reflection, I also wonder if setting to zero is as good as we can do? Could we set it to the delay such that it can only be called `now`, once all pending objections are resolved?

Reply:
I have added such a test. 

I have also grown less interested in the second point, as well, as it's verging on inter-extension awareness/interoperability, so until it's an issue, I'm inclined to leave it as-is.